posthog/temporal/batch_exports/sql.py

```diff
@@ -38,59 +13,204 @@
-"""
-
-SELECT_FROM_PERSONS_VIEW_BACKFILL = """
-SELECT
-    persons.team_id AS team_id,
-    persons.distinct_id AS distinct_id,
-    persons.person_id AS person_id,
-    persons.properties AS properties,
-    persons.person_distinct_id_version AS person_distinct_id_version,
-    persons.person_version AS person_version,
-    persons._inserted_at AS _inserted_at
-FROM
-    persons_batch_export_backfill(
-        team_id={team_id},
-        interval_end={interval_end}
-    ) AS persons
+    persons._inserted_at AS _inserted_at,
+    persons.is_deleted AS is_deleted
+FROM (
+    with new_persons as (
+        select
+            id,
+            max(version) as version,
+            argMax(_timestamp, person.version) AS _timestamp2
+        from
+            person
+        where
+            team_id = {team_id}::Int64
+            and id in (
+                select
+                    id
+                from
+                    person
+                where
+                    team_id = {team_id}::Int64
+                    and _timestamp >= {interval_start}::DateTime64
+                    AND _timestamp < {interval_end}::DateTime64
+            )
+        group by
+            id
+        having
+            (
+                _timestamp2 >= {interval_start}::DateTime64
+                AND _timestamp2 < {interval_end}::DateTime64
+            )
+    ),
+    new_distinct_ids as (
+        SELECT
+            argMax(person_id, person_distinct_id2.version) as person_id
+        from
+            person_distinct_id2
+        where
+            team_id = {team_id}::Int64
+            and distinct_id in (
+                select
+                    distinct_id
+                from
+                    person_distinct_id2
+                where
+                    team_id = {team_id}::Int64
+                    and _timestamp >= {interval_start}::DateTime64
+                    AND _timestamp < {interval_end}::DateTime64
+            )
+        group by
+            distinct_id
+        having
+            (
+                argMax(_timestamp, person_distinct_id2.version) >= {interval_start}::DateTime64
+                AND argMax(_timestamp, person_distinct_id2.version) < {interval_end}::DateTime64
+            )
+    ),
+    all_new_persons as (
+        select
+            id,
+            version
+        from
+            new_persons
+        UNION
+        ALL
+        select
+            id,
+            max(version)
+        from
+            person
+        where
+            team_id = {team_id}::Int64
+            and id in new_distinct_ids
+        group by
+            id
+    )
+    select
+        p.team_id AS team_id,
+        pd.distinct_id AS distinct_id,
+        toString(p.id) AS person_id,
+        p.properties AS properties,
+        pd.version AS person_distinct_id_version,
+        p.version AS person_version,
+        p.created_at AS created_at,
+        toBool(p.is_deleted) AS is_deleted,
```

Comment:
I'm converting this to a boolean since I figured that would be more intuitive

Reply:
I agree with this and to be honest I was surprised it is not already a boolean. Huh.